### PR TITLE
fix: display error information for failed software update due to unrecognised type

### PR DESCRIPTION
**Pull request by jarhodes314**
## Proposed changes
Don't discard the error messages from unknown software types, and specify the error message in the failure reason when only one error has occurred. This is in response to a misdiagnosed customer issue, where thin-edge gave no indication that the customer needed to create a plugin for a software type as no error message appeared anywhere in the logs.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
- #3057

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/plugin_sm/src/plugin_manager.rs [hunk 1]
```diff
@@ -245,7 +245,7 @@ impl ExternalPlugins {
             }
         }
 
-        if let Some(reason) = ExternalPlugins::error_message(error_count, command_log) {
+        if let Some(reason) = ExternalPlugins::error_message(error_count, None, command_log) {
```
Comment (albinsuresh) [inline crates/core/plugin_sm/src/plugin_manager.rs:248]: Passing hard-coded `None` here defeats the purpose, right? The error that's ignored in line no. 241 should have been passed here, right?

Comment (jarhodes314) [reply crates/core/plugin_sm/src/plugin_manager.rs:248]: True. That's an artifact of me trying to solve the install case specifically, but you're right that we shouldn't discard the error.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by albinsuresh — APPROVED** [review]
LGTM

**Pull request merged**